packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -29,6 +29,11 @@ export default function astro({ config, logging }: AstroPluginOptions): vite.Plu
 		}
 		return filename;
 	}
+	function relativeToRoot(pathname: string) {
+		const arg = startsWithForwardSlash(pathname) ? '.' + pathname : pathname;
+		const url = new URL(arg, config.projectRoot);
+		return slash(fileURLToPath(url)) + url.search;
```

Comment:
Q. Is there a difference between `slash(fileURLToPath(url))` and `url.pathname`? My understand is that `URL` would be the same as the `slash`’d version of the possibly-windows-style path.

Reply:
Yes, `url.pathname` will contain `%20` for a space whereas `fileURLToPath` removes the encoding and puts a space in. We ran into this bug before and its why the `packages/astro/test/fixtures/static build` project contains a space.